### PR TITLE
tests: add integration tests for review requests and comments

### PR DESCRIPTION
**Pull request by dcshzj**
## Problem

There are currently no integration tests for the review requests feature. This can be dangerous.

## Solution

**Breaking Changes**

- [ ] Yes - this PR contains breaking changes
- [x] No - this PR is backwards compatible

**Improvements**:

- Added some integration tests for the review requests feature

**Bug Fixes**:

- Fixed an issue with the reviews route not waiting for the results of IdentityService first, which can cause the validation check to be skipped when the user is not a member
- Added more sanity checks for comments-related functionality to prevent unauthorised users from adding or retrieving comments
- Standardised the response codes for the non-happy paths

## Tests

**Unit tests**: Run `npm run test` on this branch

## Deploy Notes

*None*

### CODE REVIEW

**Review by seaerchin — COMMENTED** [review]
lgtm w/ minor qn; skimmed over the tests but it looks fine to me. there's a few test cases identified from vapt that we should add, but that can come in another PR

File: package.json [hunk 1]
```diff
@@ -7,7 +7,7 @@
     "start": "node -r ts-node/register/transpile-only -r tsconfig-paths/register build/server.js",
     "dev:services": "docker compose up -d",
     "dev": "source .env && ts-node-dev --respawn src/server.js",
-    "test": "source .env.test && jest",
+    "test": "source .env.test && jest --runInBand",
```
[the remark this answers is not in the document]
Comment (dcshzj) [reply package.json:10]: We have some issues with the concurrency when running integration tests, as they require that the database is in a clean state before the tests are run. I think it should theoretically be possible to make the integration tests be run independently, but we will need to adjust some of the test data especially for some of our older tests.

I feel that we should rationalise the integration tests in a separate PR once all the integration tests are ready and complete (or using some method of generating fake data).

The runtime is slightly slower when the tests run sequentially, but overall it takes around 44 seconds now.

File: src/integration/Reviews.spec.ts [hunk 2]
```diff
@@ -0,0 +1,502 @@
...
+    // to ensure that the calls for .get() on the instance
+    // will actually return a value and not fail.
+    jest.mock("../services/api/AxiosInstance.ts", () => ({
+      __esModule: true, // this property makes it work
+      genericGitHubAxiosInstance: mockGenericAxios,
+    }))
+    await User.sync({ force: true })
+    await Site.sync({ force: true })
+    await Repo.sync({ force: true })
+    await SiteMember.sync({ force: true })
```
[the remark this answers is not in the document]
Comment (dcshzj) [reply src/integration/Reviews.spec.ts:134]: On first try it doesn't seem to work, will need to investigate more on why.

File: src/integration/Sites.spec.ts [hunk 3]
```diff
@@ -109,6 +109,11 @@ describe("Sites Router", () => {
 
   describe("/", () => {
     beforeAll(async () => {
+      await User.sync({ force: true })
+      await Site.sync({ force: true })
+      await Repo.sync({ force: true })
+      await SiteMember.sync({ force: true })
```
[the remark this answers is not in the document]
Comment (dcshzj) [reply src/integration/Sites.spec.ts:118]: Added comment in f83091b.

File: src/integration/Users.spec.ts [hunk 4]
```diff
@@ -55,6 +55,11 @@ const extractMobileOtp = (mobileBody: string): string =>
   mobileBody.slice(12, 12 + 6)
 
 describe("Users Router", () => {
+  beforeAll(async () => {
+    await User.sync({ force: true })
+    await Whitelist.sync({ force: true })
```
[the remark this answers is not in the document]
Comment (dcshzj) [reply src/integration/Users.spec.ts:63]: Added comment in f83091b.

**Review by seaerchin — APPROVED** [review]
lgtm

**Pull request merged**